federation/pkg/federatedtypes/hpa.go

```diff
@@ -0,0 +1,762 @@
+	}
+
+	// Average out the available current utilisation
+	if *updatedStatus.CurrentCPUUtilizationPercentage != 0 {
+		*updatedStatus.CurrentCPUUtilizationPercentage /= count
+	}
+
+	if ((fedHpa.Status.CurrentCPUUtilizationPercentage == nil &&
+		*updatedStatus.CurrentCPUUtilizationPercentage != 0) ||
+		(fedHpa.Status.CurrentCPUUtilizationPercentage != nil &&
+			*updatedStatus.CurrentCPUUtilizationPercentage !=
+				*fedHpa.Status.CurrentCPUUtilizationPercentage)) ||
+		((fedHpa.Status.LastScaleTime == nil && updatedStatus.LastScaleTime != nil) ||
+			(fedHpa.Status.LastScaleTime != nil && updatedStatus.LastScaleTime == nil) ||
+			((fedHpa.Status.LastScaleTime != nil && updatedStatus.LastScaleTime != nil) &&
+				updatedStatus.LastScaleTime.After(fedHpa.Status.LastScaleTime.Time))) ||
+		fedHpa.Status.DesiredReplicas != updatedStatus.DesiredReplicas ||
+		fedHpa.Status.CurrentReplicas != updatedStatus.CurrentReplicas {
+
+		fedHpa.Status = updatedStatus
+		_, err = a.client.AutoscalingV1().HorizontalPodAutoscalers(fedHpa.Namespace).UpdateStatus(fedHpa)
+		if err != nil {
+			return fmt.Errorf("Error updating hpa: %s status in federation: %v", fedHpa.Name, err)
+		}
+	}
+	return nil
+}
+
+// prepareForScheduling prepares the lists and totals from the
+// existing objs.
+// currentObjs has the list of all clusters, with obj as nil
+// for those clusters which do not have hpa yet.
+func prepareForScheduling(currentObjs map[string]pkgruntime.Object) (hpaLists, replicaNums, map[string]*replicaNums) {
+	lists := hpaLists{
+		availableMax: sets.NewString(),
+		availableMin: sets.NewString(),
+		noHpa:        sets.NewString(),
+	}
+	existingTotal := replicaNums{
+		min: int32(0),
+		max: int32(0),
+	}
+
+	scheduleStatus := make(map[string]*replicaNums)
+	for cluster, obj := range currentObjs {
+		if obj == nil {
+			lists.noHpa.Insert(cluster)
+			scheduleStatus[cluster] = nil
+			continue
+		}
+		if maxReplicasReducible(obj) {
+			lists.availableMax.Insert(cluster)
+		}
+		if minReplicasReducible(obj) {
+			lists.availableMin.Insert(cluster)
+		}
+
+		replicas := replicaNums{min: 0, max: 0}
+		scheduleStatus[cluster] = &replicas
+		if obj.(*autoscalingv1.HorizontalPodAutoscaler).Spec.MinReplicas != nil {
+			existingTotal.min += *obj.(*autoscalingv1.HorizontalPodAutoscaler).Spec.MinReplicas
+			replicas.min += *obj.(*autoscalingv1.HorizontalPodAutoscaler).Spec.MinReplicas
+		}
+		existingTotal.max += obj.(*autoscalingv1.HorizontalPodAutoscaler).Spec.MaxReplicas
+		replicas.max += obj.(*autoscalingv1.HorizontalPodAutoscaler).Spec.MaxReplicas
+	}
+
+	return lists, existingTotal, scheduleStatus
+}
+
+// reduceMinReplicas reduces the min replicas from existing clusters.
+// At the end of the function remainingReplicas.min should be 0 and the MinList
+// and the scheduledReplicas properly updated in place.
+func reduceMinReplicas(remainingMin int32, availableMinList sets.String, scheduled map[string]*replicaNums) int32 {
+	if remainingMin < 0 {
```

Comment:
Deferring it just means you need to remember to make the change later.  The change is so simple I'd suggest doing it now.